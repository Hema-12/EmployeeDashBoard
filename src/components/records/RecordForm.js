import React from "react";
import PropTypes from "prop-types";
import TextInput from "../common/TextInput";
import SelectInput from "../common/SelectInput";
import DatePick from "../common/DatePicker"

const RecordForm = ({
  record,
  depts,
  onSave,
  onChange,
  saving = false,
  errors = {}
}) => {
  return (
    <form onSubmit={onSave}>
      <h2>{record.id ? "Edit" : "Add"} record</h2>
      {errors.onSave && (
        <div className="alert alert-danger" role="alert">
          {errors.onSave}
        </div>
      )}
      <TextInput
        name="name"
        label="Employee Name"
        value={record.name}
        onChange={onChange}
        error={errors.name}
      />
      <TextInput
        name="manager"
        label="Manager"
        value={record.manager}
        onChange={onChange}
        error={errors.manager}
      />

      <DatePick 
        name = "doj"
        label="Date of Joining"
        value={record.doj}
        onChange={onChange} 
        className="form-control"
        error={errors.dob}
      />

      <SelectInput
        name="deptId"
        label="Dept"
        value={record.deptId || ""}
        defaultOption="Select Department"
        options={depts.map(dept => ({
          value: dept.id,
          text: dept.name
        }))}
        onChange={onChange}
        error={errors.dept}
      />

      <TextInput
        name="phno"
        label="Phone Number"
        value={record.phno}
        onChange={onChange}
        error={errors.phno}
      />

      <DatePick 
        name = "dob"
        label="Date of Birth"
        value={record.dob}
        onChange={onChange} 
        className="form-control"
        error={errors.dob}
      />

      <TextInput
        name="salary"
        label="Salary"
        value={record.salary}
        onChange={onChange}
        error={errors.salary}
      />

      <TextInput
        name="nationality"
        label="Nationality"
        value={record.nationality}
        onChange={onChange}
        error={errors.nationality}
      />

      <button type="submit" disabled={saving} className="btn btn-primary">
        {saving ? "Saving..." : "Save"}
      </button>
    </form>
  );
};

RecordForm.propTypes = {
  depts: PropTypes.array.isRequired,
  record: PropTypes.object.isRequired,
  errors: PropTypes.object,
  onSave: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  saving: PropTypes.bool
};

export default RecordForm;
