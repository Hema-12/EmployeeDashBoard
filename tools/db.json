{
  "records": [
    {
      "id": 1,
      "name": "Sampath",
      "slug": "react-auth0-authentication-security",
      "deptId": 1,
      "nationality": "Indian",
      "manager": "Supritha",
      "phno": 9560323232,
      "salary": 20000,
      "doj": "28/05/2018"
    },
    {
      "id": 2,
      "name": "Sandeep",
      "slug": "react-big-picture",
      "deptId": 1,
      "nationality": "Indian",
      "manager": "Supritha",
      "phno": 9560323232,
      "salary": 20000,
      "doj": "28/05/2018"
    },
    {
      "id": 3,
      "name": "Punnam",
      "slug": "react-creating-reusable-components",
      "deptId": 1,
      "nationality": "Indian",
      "manager": "Supritha",
      "phno": 9560323232,
      "salary": 20000,
      "doj": "28/05/2018"
    },
    {
      "id": 4,
      "name": "Hemalatha",
      "slug": "javascript-development-environment",
      "deptId": 1,
      "nationality": "Indian",
      "manager": "Supritha",
      "phno": 9560323232,
      "salary": 20000,
      "doj": "28/05/2018"
    },
    {
      "id": 5,
      "name": "Hemalatha",
      "deptId": 1,
      "nationality": "Indian",
      "manager": "Sandeep",
      "phno": "9090909090",
      "salary": "20000.00",
      "createdAt": 1559125944247,
      "slug": "hemalatha"
    },
    {
      "id": 6,
      "name": "Chandana b l",
      "deptId": 1,
      "nationality": "Indian",
      "manager": "Sandeep",
      "phno": "9090909090",
      "salary": "20000.00",
      "createdAt": 1559132189854,
      "slug": "chandana"
    },
    {
      "id": 7,
      "name": "Hemalatha",
      "deptId": 4,
      "nationality": "Indian",
      "manager": "Sandeep",
      "phno": "9090909090",
      "salary": "20000.00",
      "doj": "2019-05-15",
      "createdAt": 1559133080076,
      "slug": "hemalatha"
    },
    {
      "id": 8,
      "name": "Deepa",
      "deptId": 3,
      "nationality": "Indian",
      "manager": "Kuldeep",
      "phno": "9535162563",
      "salary": "20000.00",
      "doj": "2019-05-01",
      "dob": "1994-05-29",
      "createdAt": 1559133922684,
      "slug": "deepa"
    }
  ],
  "depts": [
    {
      "id": 1,
      "name": "APPS"
    },
    {
      "id": 2,
      "name": "BPO"
    },
    {
      "id": 3,
      "name": "HR"
    },
    {
      "id": 4,
      "name": "Finance"
    }
  ]
}