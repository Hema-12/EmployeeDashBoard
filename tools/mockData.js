const records = [
  {
    id: 1,
    name: "Sampath",
    slug: "react-auth0-authentication-security",
    deptId: 1,
    nationality: "Indian",
    manager: "Supritha",
    phno: 9560323232,
    salary: 20000.00,
    doj:"28/05/2018",
    dob:"28/05/1993"

  },
  {
    id: 2,
    name: "Sandeep",
    slug: "react-big-picture",
    deptId: 1,
    nationality: "Indian",
    manager: "Supritha",
    phno: 9560323232,
    salary: 20000.00,
    doj:"28/05/2018",
    dob:"28/05/1993"
  },
  {
    id: 3,
    name: "Punnam",
    slug: "react-creating-reusable-components",
    deptId: 1,
    nationality: "Indian",
    manager: "Supritha",
    phno: 9560323232,
    salary: 20000.00,
    doj:"28/05/2018",
    dob:"28/05/1993"
  },
  {
    id: 4,
    name: "Hemalatha",
    slug: "javascript-development-environment",
    deptId: 1,
    nationality: "Indian",
    manager: "Supritha",
    phno: 9560323232,
    salary: 20000.00,
    doj:"28/05/2018",
    dob:"28/05/1993"
  },
];

const depts = [
  { id: 1, name: "APPS" },
  { id: 2, name: "BPO" },
  { id: 3, name: "HR" },
  { id: 4, name: "Finance" }
];

const newRecord = {
  id: null,
  name: "",
  deptId: null,
  nationality: "",
  manager: "",
  phno: null,
  salary:null,
  doj:"",
  dob:"",
};

// Using CommonJS style export so we can consume via Node (without using Babel-node)
module.exports = {
  newRecord,
  records,
  depts
};
