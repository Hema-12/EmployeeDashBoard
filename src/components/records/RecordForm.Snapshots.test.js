import React from "react";
import RecordForm from "./RecordForm";
import renderer from "react-test-renderer";
import { records, depts } from "../../../tools/mockData";

it("sets submit button label 'Saving...' when saving is true", () => {
  const tree = renderer.create(
    <RecordForm
      record={records[0]}
      depts={depts}
      onSave={jest.fn()}
      onChange={jest.fn()}
      saving
    />
  );

  expect(tree).toMatchSnapshot();
});

it("sets submit button label 'Save' when saving is false", () => {
  const tree = renderer.create(
    <RecordForm
      record={records[0]}
      depts={depts}
      onSave={jest.fn()}
      onChange={jest.fn()}
      saving={false}
    />
  );

  expect(tree).toMatchSnapshot();
});
