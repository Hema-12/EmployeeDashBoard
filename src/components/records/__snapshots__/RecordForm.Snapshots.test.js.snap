// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sets submit button label 'Save' when saving is false 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <h2>
    Edit
     Record
  </h2>
  <div
    className="form-group"
  >
    <label
      htmlFor="name"
    >
      Name
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="name"
        onChange={[MockFunction]}
        type="text"
        value="Sampath"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="manager"
    >
      Manager
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="manager"
        onChange={[MockFunction]}
        type="text"
        value="Supritha"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="deptId"
    >
      Dept
    </label>
    <div
      className="field"
    >
      <select
        className="form-control"
        name="deptId"
        onChange={[MockFunction]}
        value={1}
      >
        <option
          value=""
        >
          Select Department
        </option>
        <option
          value={1}
        >
          APPS
        </option>
        <option
          value={2}
        >
          BPO
        </option>
        <option
          value={3}
        >
          HR
        </option>
        <option
          value={4}
        >
          Finance
        </option>
      </select>
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="nationality"
    >
      Nationality
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="nationality"
        onChange={[MockFunction]}
        type="text"
        value="JavaScript"
      />
    </div>
  </div>
  <button
    className="btn btn-primary"
    disabled={false}
    type="submit"
  >
    Save
  </button>
</form>
`;

exports[`sets submit button label 'Saving...' when saving is true 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <h2>
    Edit
     Record
  </h2>
  <div
    className="form-group"
  >
    <label
      htmlFor="name"
    >
      Name
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="name"
        onChange={[MockFunction]}
        type="text"
        value="Securing React Apps with Auth0"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="deptId"
    >
      Department
    </label>
    <div
      className="field"
    >
      <select
        className="form-control"
        name="deptId"
        onChange={[MockFunction]}
        value={1}
      >
        <option
          value=""
        >
          Select Department
        </option>
        <option
          value={1}
        >
          APPS
        </option>
        <option
          value={2}
        >
          BPO
        </option>
        <option
          value={3}
        >
          HR
        </option>
        <option
          value={4}
        >
          Finance
        </option>
      </select>
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="nationality"
    >
      Nationality
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="nationality"
        onChange={[MockFunction]}
        type="text"
        value="JavaScript"
      />
    </div>
  </div>
  <button
    className="btn btn-primary"
    disabled={true}
    type="submit"
  >
    Saving...
  </button>
</form>
`;
